File: backend/src/Ecom.Domain/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace Ecom.Domain.Entities;

public partial class Product
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    //if poduct name is Iphone 16 pro slug will be Iphone-16-pro which will be readable and willll be easy for the search
    public string Slug { get; set; } = null!;

    public string? ShortDescription { get; set; }

    public string? Description { get; set; }

    public decimal Price { get; set; }

    public string Sku { get; set; } = null!;

    public int StockQuantity { get; set; }

    public Guid CategoryId { get; set; }

    public Guid? BrandId { get; set; }

    public bool? IsActive { get; set; }

    public bool? IsFeatured { get; set; }

    public DateTime? CreatedAt { get; set; }
    public virtual Brand? Brand { get; set; }

    public virtual Category Category { get; set; } = null!;

    public virtual ICollection<ProductImage> ProductImages { get; set; } = new List<ProductImage>();

    public virtual ICollection<OrderItem> OrderItem { get; set; } = new List<OrderItem>();


    
    public virtual ICollection<ProductTag> ProductTags { get; set; } = new List<ProductTag>();
}

File: frontend/ecom-frontend/src/app/features/catalog/models/product.ts

// Full Product (from API)
export interface Product {
    id: string;
    name: string;
    slug: string;
    shortDescription: string | null;
    description: string | null;
    price: number;
    sku: string;
    stockQuantity: number;
    categoryId: string | null;
    categoryName: string;
    brandName: string;
    isActive: boolean;
    isFeatured: boolean;
    imageUrls: string[];
    tags: string[];
}

// Payload to send the API 
export interface AddProductPayload {
    name: string;
    shortDescription?: string;
    description?: string;
    price: number;
    sku: string;
    stockQuantity: number;
    imageUrls: string[];
    tags: string[];
    categoryName: string;
    categorySlug?: string;
    brandName: string;
    brandLogoUrl?: string;
}

//payload to update product
export interface UpdateProductPayload {
    id: string;
    name: string;
    shortDescription?: string | null;
    description?: string | null;
    price: number;
    sku: string;
    stockQuantity: number;
    categoryId: string;
    brandId?: string | null;
    brandLogoUrl?: string | null;
    isActive: boolean;
    isFeatured: boolean;
    imageUrls: string[];
    tags: string[];
}

export interface ProductForEdit {
    id: string;
    name: string;
    shortDescription?: string | null;
    description?: string | null;
    price: number;
    sku: string;
    stockQuantity: number;
    categoryId: string;
    categoryName: string;
    brandId?: string | null;
    brandName?: string | null;
    brandLogoUrl?: string | null;
    isActive: boolean;
    isFeatured: boolean;
    imageUrls: string[];
    tags: string[];
}

export interface Category{
    id:string;
    name:string;
}
export interface Brand {
    id: string;
    name: string;
}